src/vivarium_census_prl_synth_pop/components/person.py

```diff
@@ -174,20 +184,23 @@ def on_time_step(self, event: Event) -> None:
     # Helper methods #
     ##################
 
-    def _get_new_household_ids(
-            self, persons_who_move: pd.DataFrame, event: Event,
-    ) -> pd.Series:
-        households = self.population_view.subview(["household_id"]).get(event.index)
-        all_household_ids = list(households.squeeze().drop_duplicates())  # all household_ids in simulation
+    def _get_new_household_ids(self, pop: pd.DataFrame, sims_who_move: pd.Index) -> pd.Series:
+        households = pop["household_id"]
+        all_household_ids = list(
+            households.squeeze().drop_duplicates()
+        )  # all household_ids in simulation
 
-        new_household_ids = persons_who_move["household_id"].copy()  # People who move household_ids
+        new_household_ids = pop.loc[
+            sims_who_move, "household_id"
+        ].copy()  # People who move household_ids
+        old_household_ids = new_household_ids.copy()
```

Comment:
No need for this copy - you can just compare to `pop.loc[sims_who_move, "household_id"`